### PR TITLE
Add route to select random card, plus a route to the round show page …

### PR DESCRIPTION
**Pull request by kimchanyoung**
…upon completion

@jkrth617 @lksevans12 @jeder28 

I made a route to select a random card; we can add a hyperlink to that to get to the next card or something. I wasn't sure where we left off in terms of what we were going to store in the session, so lines 24 and 26 accomplish the same thing, depending on what we store in the session. If... I messed things up, sorry. :frog:

### CODE REVIEW

File: app/controllers/card.rb [hunk 1]
```diff
@@ -17,4 +17,14 @@
   new_guess = round.guesses.last.id
 
   redirect :'guesses/#{new_guess}'
+end
+
+get '/cards/random' do
+  # If the session will hold a deck
+  @card = (Deck.find(session[:deck_id]).cards - Round.find(session[:round_id]).done_cards).sample
```
Comment (jkrth617) [inline app/controllers/card.rb]: Does this sample at the end of this line happen before the subtraction? did you test this? I just want to make sure we subtract the spent cards from the total cards prior to preforming a sample and pulling out a random card. But i think your assumptions on the sessions variables are good!

File: app/controllers/card.rb [hunk 2]
```diff
@@ -17,4 +17,14 @@
...
+  # If the sessino only holds the round id
+  @card = (Round.find(session[:round_id]).guesses.first.card.deck.cards - Round.find(session[:round_id]).done_cards).sample
```
Comment (jkrth617) [inline app/controllers/card.rb:26]: i like this one less just becasue the of the Demeter violation with round going throgh guess and card to get the model ID.

**Comment by jkrth617** [comment]
I really like your @card selection on line 24. Could you comment out or delete your other technique below it and confirm that the sample happens after the subtraction? I know you are busy so when I get to DBC I will play around in IRB and make sure the order of operations works out there. :smile: :cookie: :sunglasses: :bear: :shell: 

-edit I think you do need to put parentheses around the array subtraction before the .sample call. right now the sampe is pull out a random card object form the Round's used up cards and then trying to subtract a Card obj from an array of Card objects :heavy_multiplication_x: :heavy_division_sign:  :heavy_minus_sign:

**Comment by jeder28** [comment]
Hey @kimchanyoung  we fixed some of this logic and put the card selection in the guesses controller, the first time a card is selected we just sample from the deck. I'm working on the logic for creating a round in the guess controller so we can select another card  right now :)

**Pull request closed without merging**